src/app/client/src/app/app.component.ts

```diff
@@ -122,7 +123,42 @@ export class AppComponent implements OnInit, OnDestroy {
           _.get(this.activatedRoute, 'snapshot.firstChild.firstChild.firstChild.data.hideHeaderNFooter');
       });
   }
+  public makeUTMSession(params) {
+    for (const item in params) {
+      if (params.hasOwnProperty(item)) {
+        switch (item) {
+          case ('channel'):
+            this.resultJson.push({ 'id': params[item], 'type': 'Source' });
+            break;
+          case ('utm_campaign'):
+            this.resultJson.push({ 'id': params[item], 'type': 'Source' });
+            break;
+          case 'utm_medium':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmMedium' });
+            break;
+          case 'utm_source':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmSource' });
+            break;
+          case 'utm_term':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmTerm' });
+            break;
+          case 'utm_content':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmContent' });
+            break;
+          default:
+            break;
+        }
+      }
+    }
+    sessionStorage.setItem('UTM', JSON.stringify(this.resultJson));
```

Comment:
If you set this in session when will these be cleared. This param will get added to telemetry for all user action. If user come via explore page second time, then also telemetry will have utm params right. please check this cases.

Reply:
Yes until and unless you explicitly logout or close the tab session won't clear.

Reply:
ok, please check all cases once. Before going with session